test/unit/org/apache/cassandra/cql3/ast/Bind.java

```diff
@@ -47,7 +47,7 @@ public AbstractType<?> type()
 
     public ByteBuffer encode()
     {
-        return ((AbstractType) type).decompose(value);
+        return value instanceof ByteBuffer ? (ByteBuffer) value : ((AbstractType) type).decompose(value);
```

Comment:
the next patch adds a model, and to keep things simple all values are `ByteBuffer` right away, and not the java objects `AbstractType` expects, but that broke the `ast` package's `Value` types... so needed to explicitly handle in both types